### PR TITLE
[jit] Shape propagation batch_norm, dropout, quantize, hardswidh

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64751
* #64750
* #64741
* __->__ #64740

Differential Revision: [D30839111](https://our.internmc.facebook.com/intern/diff/D30839111)

Adding shape propagation functions to handle mobilenetv3-large (base and optimized)
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

### CODE REVIEW

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839111).

**Review by eellison — COMMENTED** [review]
if you rebase, you can just register these as `unary` now after https://github.com/pytorch/pytorch/pull/65095

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839111).

File: torch/csrc/jit/runtime/symbolic_shape_registry.cpp [hunk 1]
```diff
@@ -436,6 +436,12 @@ const std::string shape_compute_functions =
           assert len(input) == 4
           return conv_output_size(input, weight, bias, stride, padding, dilation, groups)
 
+        def batch_norm(input: List[int], weight: List[int], bias: Optional[List[int]], running_mean: Optional[List[int]], running_var: Optional[List[int]], training: bool, momentum: float, eps: float, cudnn_enabled: bool):
```
Comment (eellison) [inline torch/csrc/jit/runtime/symbolic_shape_registry.cpp:439]: Are we not using unary here to distinguish between pointwise/elementwise ops ?

**Review by eellison — APPROVED** [review]
lgtm, question about batch_norm

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839111).
